Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -83,10 +93,25 @@ private async Task<ReadHttpResponseMessage> SendAsync<TRequest>(RestMethod restM
             }
             httpRequest.Headers.Add(VersionHeaderBuilder.HeaderName, RestVersion);
 
-            // TODO: How do we cancel this?
-            await AddAuthHeadersAsync(httpRequest, restMethod).ConfigureAwait(false);
+            await AddAuthHeadersAsync(httpRequest, restMethod, options, deadlineToken).ConfigureAwait(false);
+ 
+            HttpResponseMessage httpResponseMessage;
+            using (CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(options.CancellationToken, deadlineToken))
+            {
+                try
+                {
+                    httpResponseMessage = await _httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseContentRead, linkedCts.Token).ConfigureAwait(false);
+                }
+                catch(TaskCanceledException)
```

Comment:
Nit: space after catch. But actually, we can simplify this anyway:

```csharp
catch (TaskCanceledException) when (deadline.IsCancellationRequested)
{
    throw new RpcException(new Status(StatusCode.DeadlineExceeded, $"The timeout was reached when calling a method `{restMethod.FullName}`"));
}
```

No need to rethrow explicitly.

Reply:
👍